bin/cheritest/cheritest_fault.c

```diff
@@ -210,10 +210,6 @@ test_nofault_cfromptr(const struct cheri_test *ctp __unused)
 	void * __capability cb; /* derived from here */
 	int rt;
 
-	/*
-	 * XXXRW: Could we be using CHERI_CFROMPTR() here to avoid explicit
-	 * inline assembly?
-	 */
```

Comment:
We should probably use `cheri_cap_from_pointer` (which wraps `__builtin_cheri_cap_from_pointer`) here?

Reply:
Quite possibly.  I'm sure this predates compiler builtins.  I was only removing a comment to the macro that's being removed, but I think it would be useful to use compiler builtins instead.  For example, could we replace 'cheri_ptr(buf, 256)' with just 'cb = buf', or are you suggesting just using '(void)cheri_cap_from_pointer(buf, 10)'?

Reply:
`cheri_ptr` comes from sys/cheri/cheric.h, and turns a pointer+length into a bounded cap, ie a CFromPtr plus CSetBounds, without using any assembly. If you're removing the comment but not modifying the assembly referenced, I would instead suggest you *replace* the comment with one referencing `cheri_cap_from_pointer`.

Reply:
Note that `cheri_cap_from_pointer` takes a pointer and an explicit authorising capability, whereas `cheri_ptr` is always DDC-derived.

Reply:
Does the updated comment express what you want?

Reply:
Yes, thanks